### DISCUSSION TITLE
Angular 13 (Typescript 4.4) ?

### DISCUSSION THREAD

**Discussion by yekurtal** [timostamm/protobuf-ts#245] [General]
Hello love the project,

At the moment I am using this project with Angular 11 and I love it.

But 11 support is ending come [May 11, 2022](https://angular.io/guide/releases#support-policy-and-schedule).

Upgrading to support Angular 13 would require to upgrade upgrade TypeScript to version 4.4 as [Angular 13 dropped support for anything lower then 4.4.2](https://github.com/angular/angular/pull/43642).

Are there any plans to upgrade this project to Typescript 4.4?

**Comment by timostamm** [comment]
Hey Yunus, the plugin uses an older version of TypeScript to generate code, but you should be able to compile the generated code with a newer version of TypeScript just fine! 

I just verified this in [test-generated](https://github.com/timostamm/protobuf-ts/tree/master/packages/test-generated) with TypeScript 4.6.2. I had to update a few spec files and conformance_ts.ts for cases where TypeScript has become more strict, but didn't touch the generated code at all.

Depending on your tsconfig.json, you might get errors from the generated code that we cannot solve without breaking older TypeScript versions. For this very reason, you can set the plugin option `--ts_opt ts_nocheck`, which will add a comment to each generated file that tells TypeScript not to type-check. This is not ideal, but it seems to be the best option available, and the next major release will enable `ts_nocheck` by default.

So using this plugin with a recent TypeScript version should be fine.

The only potential problem I am aware of is using the generated code in a ECMAScript module project. [We should be generating a `.js` file extension to imports](https://github.com/timostamm/protobuf-ts/issues/182) to better support this, but this hasn't been implemented yet. I do not believe this is an issue for Angular, but in case you notice something, let me know.

**Reply by yekurtal** [reply]
Hey Timo,

Thank you for the quick reply, I got it working with Angular 13.2.5 by using the `--force` when installing `@protobuf-ts/runtime-angular`. Would you accept a PR if I added a comment about it in the manual?

I do have one more question: is the plugin option [`client_rx`](https://github.com/timostamm/protobuf-ts/pull/10/files#diff-201a2f185bb0bd20fc824a4903c2069ce0e65809bed814e5bffb3c49fb2ff888R128) deprecated? Because I couldn't find anything about it in the changeling.

EDIT:
I have found the the commit where you [drop client styles](https://github.com/timostamm/protobuf-ts/commit/29b2f7ec9cbfd4f7f180675510b91b11959525a5#diff-201a2f185bb0bd20fc824a4903c2069ce0e65809bed814e5bffb3c49fb2ff888L149), could you please tell me how one would still use rxjs in the generated client? Wrap everything in observables?

**Reply by timostamm** [reply]
Ah, we need to relax the angular version constraints here:

https://github.com/timostamm/protobuf-ts/blob/41faab08b42cd459ce7f70faaa0811daa9b10541/packages/runtime-angular/package.json#L35-L36

Then it shouldn't be necessary to use `--force`. Thanks for raising this!

RXJS: First-class support for RX has been important to me, but the way it was implemented was difficult to test and maintain, so it was pulled in version [2.0.0-alpha.16](https://github.com/timostamm/protobuf-ts/blob/master/CHANGELOG.md#v200-alpha16). The best alternative for now is wrapping in observables with RXJS' `from()`.

**Comment by timostamm** [comment]
Thanks for testing, Yunus. Release [v2.2.4](https://github.com/timostamm/protobuf-ts/releases/tag/v2.2.4) has relaxed the compatibility constraints to Angular version 8 to 13.
